### PR TITLE
Initial Hidden Fates Implementation

### PR DESCRIPTION
**Pull request by luongthomas**
This PR includes work for nearly all the Pokemon and some of the Trainer cards.

I've added a //TODO: tag for the ones that still need work.

### CODE REVIEW

File: src/tcgwars/logic/impl/gen7/HiddenFates.groovy [hunk 1]
```diff
@@ -1058,83 +1148,98 @@ public enum HiddenFates implements CardInfo {
...
       };
       case SABRINA_S_SUGGESTION_65:
-      return supporter (this) {
-        text "Your opponent reveals their hand. You may choose a Supporter card you find there and use the effect of that card as the effect of this card."
-        onPlay {
-        }
-        playRequirement{
-        }
-      };
+      return copy(SABRINAS_SUGGESTION_154, this);
```
Comment (axpendix) [inline src/tcgwars/logic/impl/gen7/HiddenFates.groovy]: This needs to be prefixed with `TeamUp.`

File: src/tcgwars/logic/impl/gen7/HiddenFates.groovy [hunk 2]
```diff
@@ -1016,39 +1103,42 @@ public enum HiddenFates implements CardInfo {
...
       case BLAINE_S_LAST_STAND_52:
-      return supporter (this) {
-        text "You can play this card only when it is the last card in your hand." +
-          "Draw 2 cards for each [R] Pokémon you have in play."
-        onPlay {
-        }
-        playRequirement{
-        }
-      };
+      return copy (BLAINE_S_LAST_STAND_58, this);
```
Comment (axpendix) [inline src/tcgwars/logic/impl/gen7/HiddenFates.groovy]: `copy(DragonMajesty.BLAINE_S_LAST_STAND_58, this)`

File: src/tcgwars/logic/impl/gen7/HiddenFates.groovy [hunk 3]
```diff
@@ -1016,39 +1103,42 @@ public enum HiddenFates implements CardInfo {
...
       case BROCK_S_GRIT_53:
-      return supporter (this) {
-        text "Shuffle 6 in any combination of Pokémon and basic Energy cards from your discard pile into your deck."
-        onPlay {
-        }
-        playRequirement{
-        }
-      };
+      return copy (BROCKS_GRIT_135, this)
```
Comment (axpendix) [inline src/tcgwars/logic/impl/gen7/HiddenFates.groovy]: `TeamUp.`

File: src/tcgwars/logic/impl/gen7/HiddenFates.groovy [hunk 4]
```diff
@@ -1058,83 +1148,98 @@ public enum HiddenFates implements CardInfo {
...
       };
       case GIOVANNI_S_EXILE_57:
-      return supporter (this) {
-        text "Discard up to 2 of your Benched Pokémon that have no damage counters on them and all cards attached to them."
-        onPlay {
-        }
-        playRequirement{
-        }
-      };
+      return copy(GIOVANNI_S_EXILE_174, this);
```
Comment (axpendix) [inline src/tcgwars/logic/impl/gen7/HiddenFates.groovy]: `UnbrokenBonds.`

File: src/tcgwars/logic/impl/gen7/HiddenFates.groovy [hunk 5]
```diff
@@ -1058,83 +1148,98 @@ public enum HiddenFates implements CardInfo {
...
       };
       case KOGA_S_TRAP_59:
-      return supporter (this) {
-        text "Your opponent's Active Pokémon is now Confused and Poisoned."
-        onPlay {
-        }
-        playRequirement{
-        }
-      };
+      return copy(KOGA_S_TRAP_177, this);
```
Comment (axpendix) [inline src/tcgwars/logic/impl/gen7/HiddenFates.groovy]: `UnbrokenBonds.`

File: src/tcgwars/logic/impl/gen7/HiddenFates.groovy [hunk 6]
```diff
@@ -1058,83 +1148,98 @@ public enum HiddenFates implements CardInfo {
...
       case LT_SURGE_S_STRATEGY_60:
-      return supporter (this) {
-        text "You can play this card only if you have more Prize cards remaining than your opponent." +
-          "During this turn, you can play 3 Supporter cards (including this card)."
-        onPlay {
-        }
-        playRequirement{
-        }
-      };
+      return copy(LT_SURGE_S_STRATEGY_178, this);
```
Comment (axpendix) [inline src/tcgwars/logic/impl/gen7/HiddenFates.groovy]: `UnbrokenBonds.`

**Comment by axpendix** [comment]
Hi Thomas, amazing work there, thank you! 
I've just mentioned a couple of missing references. Please complete them then merge this PR. Afterwards, I will try to complete TODOs, then we'll release it.

**Comment by luongthomas** [comment]
> Hi Thomas, amazing work there, thank you!
> I've just mentioned a couple of missing references. Please complete them then merge this PR. Afterwards, I will try to complete TODOs, then we'll release it.

Alright, I've made the fixes and I'll merge them.  Thanks for reviewing!

**Comment by axpendix** [comment]
I need to add HIF to the internal collection class for it to be tested on dev ui. Sorry that I couldn’t tackle this friction point yet.

**Pull request merged**